nemo/collections/asr/modules/conformer_encoder.py

```diff
@@ -127,14 +127,29 @@ def input_example(self, max_batch=1, max_dim=256):
             A tuple of input examples.
         """
         dev = next(self.parameters()).device
-        input_example = torch.randn(max_batch, self._feat_in, max_dim).to(dev)
-        input_example_length = torch.randint(1, max_dim, (max_batch,)).to(dev)
-
-        if hasattr(self, 'export_cache_support') and self.export_cache_support:
-            cache_last_channel = torch.randn(self.n_layers, max_batch, max_dim, self.d_model).to(dev)
-            cache_last_time = torch.randn(self.n_layers, max_batch, self.d_model, self.conv_context_size[0]).to(dev)
-            all_input_example = tuple([input_example, input_example_length, cache_last_channel, cache_last_time])
+        if self.export_cache_support:
+            window_size = max_dim
+            if self.streaming_cfg is not None:
+                if isinstance(self.streaming_cfg.chunk_size, list):
+                    chunk_size = self.streaming_cfg.chunk_size[1]
+                else:
+                    chunk_size = self.streaming_cfg.chunk_size
+                if isinstance(self.streaming_cfg.pre_encode_cache_size, list):
+                    pre_encode_cache_size = self.streaming_cfg.pre_encode_cache_size[1]
+                else:
+                    pre_encode_cache_size = self.streaming_cfg.pre_encode_cache_size
+                window_size = chunk_size + pre_encode_cache_size
+            input_example = torch.randn(max_batch, self._feat_in, window_size, device=dev)
+            input_example_length = torch.full((max_batch,), window_size, device=dev, dtype=torch.int32)
```

Comment:
Why full is being used instead of randint?

Reply:
fixed